cmd/gitserver/server/patch.go

```diff
@@ -217,11 +218,12 @@ func (s *Server) createCommitFromPatch(ctx context.Context, req protocol.CreateC
 		committerEmail = authorEmail
 	}
 
-	cmd = exec.CommandContext(ctx, "git", "commit", "-m", fmt.Sprintf("%q", message))
-	if req.Gerrit != nil {
-		// The Change ID needs to be the footer of the commit for Gerrit.
-		cmd = exec.CommandContext(ctx, "git", "commit", "-m", fmt.Sprintf("%q", message), "-m", fmt.Sprintf("Change-Id: %s", req.Gerrit.ChangeID))
+	formattedMessages := []string{"commit"}
+	for _, m := range req.CommitInfo.Messages {
+		formattedMessages = append(formattedMessages, "-m", m)
```

Comment:
Ought we still quote each part of the commit message, as before? Otherwise I think certain characters like dashes in the commit message can get misinterpreted by the shell as additional command arguments and cause errors:

```suggestion
		formattedMessages = append(formattedMessages, "-m", fmt.Sprintf("%q", m))
```

Reply:
It seems like this change causes it not to parse the Change ID correctly because of the quotes, so I think maybe we can move this formatting to the BuildCommitOpts and then just not do it for the Change ID message in Gerrit's Build commit opts? what do you think?

Reply:
Huh interesting! Can you tell why it can't parse the Change ID part of the commit message exactly? I'm okay doing that if we really have to, it just feels a bit bad putting that responsibility on the consumer.